src/huggingface_hub/repocard_data.py

```diff
@@ -514,12 +516,22 @@ def eval_results_to_model_index(
     # Here, we make a map of those pairs and the associated EvalResults.
     task_and_ds_types_map = defaultdict(list)
     for eval_result in eval_results:
-        task_and_ds_pair = (eval_result.task_type, eval_result.dataset_type)
+        task_and_ds_pair = (
+            eval_result.task_type,
```

Comment:
The unique combination for evaluation results is `(task, dataset, config, split)`, so we extend the logic here to avoid erasing this information from the evaluation results

Reply:
Nice solving this bug @lewtun ! Do you see any way we could avoid a break in the future if another field is added to `EvalResult` ? Or is it most likely not gonna happen ?

Reply:
Good idea! Added in [8e4eae2](https://github.com/huggingface/huggingface_hub/pull/1214/commits/8e4eae2792c9107ba2ffd7486cda9729906e24c1)